src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1667,6 +1830,829 @@ class Simplex_tree {
+  *
+  * @param[out]    a range whose iterators are of value_type Simplex_handle.
+  *
+  * A ZigzagEdgeRange must be a range of ZigzagEdge. A model of ZigzagEdge
+  * must contain operations:
+  * Vertex_handle u() returns the endpoint with smaller label for <,
+  * Vertex_handle v() returns the endpoint with larger label for <,
+  * for a vertex with label x, u() == v() == x.
+  * Filtration_value fil() returns the filtration value in the zigzag
+  * filtration,
+  * bool type() returns true if the vertex/edge is inserted, false if it is removed.
+  *
+  * Zigzag_edge< Simplex_tree > is a model of ZigzagEdge.
+  */
+  template< class ZigzagEdgeRange >
+  Zigzagfiltration_simplex_range
+  zigzag_simplex_range( ZigzagEdgeRange &zz_edge_fil, int dim_max )
+  {
+    return
+      Zigzagfiltration_simplex_range(
+          Zigzagfiltration_simplex_iterator(this, zz_edge_fil, dim_max)
+          , Zigzagfiltration_simplex_iterator()  );
+  }
+/** \brief Returns a range of simplices for the (dim_max)-skeleton of the 
+  * oscillating Rips zigzag filtration with parameters mu and nu, built on top 
+  * of the point cloud 'points' with distance function 'distance'.
+  *
+  * @param[in] nu,mu    scale parameters for the oscillating Rips zigzag filtration.
+  * @param[in] dim_max  maximal dimension of a simplex.
+  * @param[in] points   range over a set of Points
+  * @param[in] distance a distance functions on Points
+  * @param[in] order_policy a policy for reordering the points in 'points'. Can be 
+  * of type 'already_ordered', 'farthest_point_ordering', or 'random_point_ordering'
+  */
+  template< typename PointRange, typename Distance, typename OrderPolicy >
+  Zigzagfiltration_simplex_range
+  zigzag_simplex_range( Filtration_value const          nu,
+                        Filtration_value const          mu,
+                        int                             dim_max, 
+                        PointRange       const        & points,
+                        Distance         const          distance,
+                        OrderPolicy                     order_policy)
+  {
+    return
+      Zigzagfiltration_simplex_range(
+          Zigzagfiltration_simplex_iterator(this, nu, mu, dim_max, points, distance, order_policy) , Zigzagfiltration_simplex_iterator()  );
+  }
+
+public:
+/** \brief Initializes a filtration range for the (dim_max)-skeleton of a 
+  * flag zigzag 
+  * filtration represented a by a sequence of insertions and 
+  * deletions of vertices and edges.
+  *
+  * @param[in] zz_edge_fil         range of Zigzag_edge< Simplex_tree >,
+  *                                containing a valid sequence of insertions and
+  *                                removal of vertices and edges.
+  *
+  * @param[in] dim_max             maximal dimension of the dynamic flag complex
+  *                                constructed.
+  *
+  * @param[out]    a range whose iterators are of value_type Simplex_handle.
```

Comment:
```suggestion
  * @return    a range whose iterators are of value_type Simplex_handle.
```
Doxygen complains about this